### PR TITLE
feat: add new database microservice API URL to the frontend

### PR DESCRIPTION
**Pull request by rhodinemma**
# Description

The PR creates provision for the database API and modifies the database page functionality to interact with the database microservice API

## Type of change.

- [x] New feature (non-breaking change which adds functionality)

## Trello Ticket ID

https://trello.com/c/aBnLDgou

## How Can This Been Tested?

Run this branch locally and checkout the `http://localhost:3000/projects/<projectID>/databases/`, validate the functionalities below

- [x] Retrieving a list of databases for that user
- [x] Creating a database 
- [x] Viewing details for a single database
- [x] Modifying the database's password
- [x] Disabling/Enabling a database
- [x] Deleting a database   

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

# Screenshots

Screenshot 2024-05-22 at 7 56 14 PM

### CODE REVIEW

**Review by Mubangizi — COMMENTED** [review]
Hey @rhodinemma, please merge into `microservice` branch, not `develop` 🙏

**Comment by Mubangizi** [comment]
I have tried testing this out but I can't seem to create any database, the page just reloads and nothing happens, any idea on what could be causing this?

File: src/pages/DBSettingsPage/index.jsx [hunk 1]
```diff
@@ -289,7 +291,7 @@ class DBSettingsPage extends React.Component {
     this.setState({
       gettingPassword: true,
     });
-    handleGetRequest(`/projects/${projectID}/databases/${databaseID}/password`)
+    handleGetRequest(`${DATABASE_API_URL}/databases/${databaseID}/password`)
```
Comment (Mubangizi) [inline src/pages/DBSettingsPage/index.jsx]: You could modify `handleGetRequest` implementation to something like 
```
export const handleGetRequest = (endpoint, axiosInstance = axios) => {
  return new Promise((resolve, reject) => {
    axiosInstance
      .get(endpoint)
      .then((response) => {
        resolve(response);
      })
      .catch((error) => {
        reject(error);
      });
  });
};
```
we are making it have an optional argument, for an Axios instance
```suggestion
    handleGetRequest(`${DATABASE_API_URL}/databases/${databaseID}/password`, databaseAxios)
```

**Review by Mubangizi — CHANGES_REQUESTED** [review]
For matters of configuration and independency, please create a `database_axios` instance and have it handle the **database API** requests. 
Having the backend axios instance handle database requests might become problematic in future.

You can base off [this](https://github.com/crane-cloud/frontend/pull/818/files#diff-3b1ad3ad19f985c72e3a08852a582a03e5f275b811d0ffba5c396c306c786419) to see how that can be done.

Then you can do further modifications to some of the functions like `handleGetRequest`

**Comment by rhodinemma** [comment]
> I have tried testing this out but I can't seem to create any database, the page just reloads and nothing happens, any idea on what could be causing this?

That screenshot in the PR ain't magic...i was testing with my local database API instance.
i haven't had time but from our last conversation i requested you to validate the deployment of the database API since it didn't have the latest changes since you are the one who setup the deployment 👍🏾

File: src/pages/DBSettingsPage/index.jsx (hunk 1 above)
Comment (rhodinemma) [reply src/pages/DBSettingsPage/index.jsx]: sounds good 👍🏾

**Comment by khalifan-kfan** [comment]
[https://github.com/crane-cloud/frontend/blob/774834dd4d4315b7b1140d175b305135cef6d48c/.github/workflows/microservice.yml#L62](Link)

@rhodinemma  please pick new changes from the microservice branch and add  the environment variable in the file like shown in the link above to help with the deployments

**Comment by rhodinemma** [comment]
> [https://github.com/crane-cloud/frontend/blob/774834dd4d4315b7b1140d175b305135cef6d48c/.github/workflows/microservice.yml#L62](Link)
> 
> @rhodinemma please pick new changes from the microservice branch and add the environment variable in the file like shown in the link above to help with the deployments

okay please, thank you

**Comment by Mubangizi** [comment]
> > I have tried testing this out but I can't seem to create any database, the page just reloads and nothing happens, any idea on what could be causing this?
> 
> That screenshot in the PR ain't magic...i was testing with my local database API instance. i haven't had time but from our last conversation i requested you to validate the deployment of the database API since it didn't have the latest changes since you are the one who setup the deployment 👍🏾

Okay, so when should we look into this?

**Comment by rhodinemma** [comment]
> > > I have tried testing this out but I can't seem to create any database, the page just reloads and nothing happens, any idea on what could be causing this?
> > 
> > 
> > That screenshot in the PR ain't magic...i was testing with my local database API instance. i haven't had time but from our last conversation i requested you to validate the deployment of the database API since it didn't have the latest changes since you are the one who setup the deployment 👍🏾
> 
> Okay, so when should we look into this?

today 👍🏾

**Review by Mubangizi — APPROVED** [review]
LGTM

**Pull request merged**